src/syscheckd/src/db/src/registry.cpp

```diff
@@ -21,202 +21,50 @@
-    {
-        //FIMDB::instance().loggingFunction(LOG_ERROR, ex.what());
-        return FIMDB_ERR;
-    }
-}
-
-fim_registry_value_data* fim_db_get_registry_data(unsigned int key_id, const char* name)
-{
-
-    nlohmann::json dbItem;
-    std::string strName(name);
-
-    std::string rawFilter = "WHERE name = " + strName + " AND key_id = " + std::to_string(key_id);
-    nlohmann::json columnList;
-    columnList["column_list"] = "[*]";
-
-    nlohmann::json query = FIMDBHelper::dbQuery(std::string(FIMDB_REGISTRY_VALUE_TABLENAME), columnList, rawFilter, "");
-
-    try
-    {
-        FIMDBHelper::getDBItem<FIMDB>(dbItem, query);
-    }
-    catch (std::exception& ex)
-    {
-        FIMDB::instance().loggingFunction(LOG_ERROR, ex.what());
-        return 0;
-    }
-
-    RegistryValue reg_value(dbItem);
-    return (reg_value.toFimEntry())->registry_entry.value;
-}
-
-fim_registry_key* fim_db_get_registry_key(const char* path, unsigned int arch)
-{
-    nlohmann::json dbItem;
-    std::string strPath(path);
-
-    std::string rawFilter = "WHERE path = " + strPath + " AND arch = " + std::to_string(arch);
-    nlohmann::json columnList;
-    columnList["column_list"] = "[*]";
-
-    nlohmann::json query = FIMDBHelper::dbQuery(std::string(FIMDB_REGISTRY_KEY_TABLENAME), columnList, rawFilter, "");
-
-    try
-    {
-        FIMDBHelper::getDBItem<FIMDB>(dbItem, query);
-    }
-    catch (std::exception& ex)
-    {
-        FIMDB::instance().loggingFunction(LOG_ERROR, ex.what());
-        return 0;
-    }
-
-    RegistryKey reg_key(dbItem);
-    return reg_key.toFimEntry()->registry_entry.key;
-
-}
-
-fim_registry_key* fim_db_get_registry_key_using_id(unsigned int key_id)
-{
-    nlohmann::json dbItem;
-
-    std::string rawFilter = "WHERE item_id = " + std::to_string(key_id);
-    nlohmann::json columnList;
-    columnList["column_list"] = "[*]";
-
-    nlohmann::json query = FIMDBHelper::dbQuery(std::string(FIMDB_REGISTRY_KEY_TABLENAME), columnList, rawFilter, "");
-
-    try
-    {
-        FIMDBHelper::getDBItem<FIMDB>(dbItem, query);
-    }
-    catch (std::exception& ex)
-    {
-        FIMDB::instance().loggingFunction(LOG_ERROR, ex.what());
-        return 0;
-    }
-
-    RegistryKey reg_key(dbItem);
-    return reg_key.toFimEntry()->registry_entry.key;
-}
-
 int fim_db_get_count_registry_key()
 {
-    int res = 0;
+    auto count { 0 };
 
     try
     {
-        FIMDBHelper::getCount<FIMDB>(std::string(FIMDB_REGISTRY_KEY_TABLENAME), res);
+        count = DB::instance().countEntries(FIMDB_REGISTRY_KEY_TABLENAME, COUNT_SELECT_TYPE::COUNT_ALL);
     }
-    catch (std::exception& ex)  // EMPTY_TABLE_METADATA
+    // LCOV_EXCL_START
+    catch (const std::exception& err)
     {
-        FIMDB::instance().loggingFunction(LOG_ERROR, ex.what());
-        return FIMDB_ERR;
+        FIMDB::instance().logFunction(LOG_ERROR, err.what());
     }
 
-    return res;
+    // LCOV_EXCL_STOP
+
+    return count;
 }
 
 int fim_db_get_count_registry_data()
```

Comment:
This function needs to documentation inside db.h

Reply:
Already have documentation placed in db.h